### PR TITLE
Feat(web): Extend `FileUploaderAttachment` with preview and custom actions

### PR DESCRIPTION
**Pull request by adamkudrna**
## Description

- New options for `FileUploaderAttachment`:
  - preview image
  - slot for custom actions or other elements
- Mark interactive examples in `FileUploader` demo so it's clear when our JS plugin is involved and when it's not.

### Additional context

image

### Issue reference

Q3 target, no implementation issue yet.

### CODE REVIEW

File: packages/web/src/scss/components/FileUploader/README.md [hunk 1]
```diff
@@ -302,6 +309,128 @@ up the template and apply it on any attachments the user wants to upload.
...
+
+To attach functionality, you need to do so after the `queuedFile.fileUploader` event has been fired:
+
+```js
+const editHandler = (event) => {
+  alert('Edit button clicked');
+};
+
+document.addEventListener('queuedFile.fileUploader', () => {
+  document.querySelectorAll('[data-element="edit"]').forEach((element) => {
```
Comment (literat) [inline packages/web/src/scss/components/FileUploader/README.md:398]:
```suggestion
  document.querySelectorAll('[data-spirit-element="edit"]').forEach((element) => {
```

File: packages/web/src/scss/components/FileUploader/README.md [hunk 2]
```diff
@@ -302,6 +309,128 @@ up the template and apply it on any attachments the user wants to upload.
...
+```js
+let filesQueueCounter = 0;
+
+const editHandler = (event) => {
+  alert('Edit button clicked');
+};
+
+document.addEventListener('queuedFile.fileUploader', () => {
+  filesQueueCounter++;
+  document.querySelectorAll('[data-element="edit"]').forEach((element) => {
```
Comment (literat) [inline packages/web/src/scss/components/FileUploader/README.md:416]:
```suggestion
  document.querySelectorAll('[data-spirit-element="edit"]').forEach((element) => {
```

File: packages/web/src/scss/components/FileUploader/README.md [hunk 3]
```diff
@@ -302,6 +309,128 @@ up the template and apply it on any attachments the user wants to upload.
...
+    element.removeEventListener('click', editHandler); // Prevent multiple event listeners
+    element.addEventListener('click', editHandler);
+  });
+});
+
+document.addEventListener('unqueueFile.fileUploader', () => {
+  filesQueueCounter--;
+  if (filesQueueCounter === 0) {
+    document.querySelectorAll('[data-element="edit"]').forEach((element) => {
```
Comment (literat) [inline packages/web/src/scss/components/FileUploader/README.md:425]:
```suggestion
    document.querySelectorAll('[data-spirit-element="edit"]').forEach((element) => {
```

File: packages/web/src/scss/components/FileUploader/README.md (hunk 1 above)
Comment (adamkudrna) [reply packages/web/src/scss/components/FileUploader/README.md:398]: This is just an example, not a part of Spirit code base.

**Review by crishpeen — APPROVED** [review]

File: packages/web/src/scss/components/FileUploader/README.md (hunk 1 above)
Comment (literat) [reply packages/web/src/scss/components/FileUploader/README.md]: Ok.

File: packages/web/src/scss/components/FileUploader/index.html [hunk 4]
```diff
@@ -440,30 +455,173 @@ <h3 id="attachments-fileUploadDisabled" hidden>Attachments</h3>
...
+            <use xlink:href="/icons/svg/sprite.svg#close" />
+          </svg>
+        </button>
+      </li>
+      <!-- FileUploaderAttachment: end -->
+
+      <!-- FileUploaderAttachment: start -->
+      <li class="FileUploaderAttachment">
+        <span class="FileUploaderAttachment__image">
+          <img src="http://placekitten.com/200/300" alt="" />
```
Comment (literat) [inline packages/web/src/scss/components/FileUploader/index.html:490]: Just for consistency: Can we please use only one service for example images? In https://spirit-design-system-demo.netlify.app/src/scss/helpers/images/ we are using picsum.photos. I think we should pick only one service for these examples.

**Review by literat — APPROVED** [review]
LGTM and thanks for the "visual demo only" tags. They are looking very good. 👍

File: packages/web/src/scss/components/FileUploader/index.html (hunk 4 above)
Comment (adamkudrna) [reply packages/web/src/scss/components/FileUploader/index.html:490]: obrazek

File: packages/web/src/scss/components/FileUploader/README.md [hunk 5]
```diff
@@ -322,6 +329,134 @@ up the template and apply it on any attachments the user wants to upload.
...
+</span>
+```
+
+Full example:
+
+```html
+<li class="FileUploaderAttachment">
+  <!-- Preview image: start -->
+  <span class="FileUploaderAttachment__image">
+    <img src="http://placekitten.com/200/300" alt="" />
```
[the remark this answers is not in the document]
Comment (adamkudrna) [reply packages/web/src/scss/components/FileUploader/README.md:348]: OK, I used the file name.

**Pull request merged**